### PR TITLE
Deprecate certain frequency strings

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR deprecates "H", "N", "T", "L", "U" and "S" as frequencies in all datetime APIs. This PR prepares `branch-24.04` for `pandas-2.2` support.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_resampling.py [hunk 1]
```diff
@@ -24,51 +24,51 @@ def assert_resample_results_equal(lhs, rhs, **kwargs):
...
     assert_resample_results_equal(
-        psr.resample("3T").sum(),
-        gsr.resample("3T").sum(),
+        psr.resample("3min").sum(),
+        gsr.resample("3min").sum(),
     )
 
 
-@pytest.mark.parametrize("rule", ["2S", "10S"])
+@pytest.mark.parametrize("rule", ["2s", "10"])
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_resampling.py:49]:
```suggestion
@pytest.mark.parametrize("rule", ["2s", "10s"])
```

File: python/cudf/cudf/_lib/datetime.pyx [hunk 2]
```diff
@@ -85,19 +87,34 @@ cdef libcudf_datetime.rounding_frequency _get_rounding_frequency(object freq):
...
+        warnings.warn(
+            f"FutureWarning: {freq} is deprecated and will be "
+            "removed in a future version, please use "
+            f"{old_to_new_freq_map[freq]} instead.",
+            FutureWarning
+        )
     if freq == "D":
         freq_val = libcudf_datetime.rounding_frequency.DAY
-    elif freq == "H":
+    elif freq in ("H", "h"):
```
Comment (mroeschke) [inline python/cudf/cudf/_lib/datetime.pyx:107]: Could we do `freq = old_to_new_freq_map.get(freq, freq)` and then update these `elif` checks with the future frequency?

Comment (galipremsagar) [reply python/cudf/cudf/_lib/datetime.pyx]: Yup, updated the code 👍

**Review by mroeschke — APPROVED** [review]

**Pull request merged**